src/interfaces/tadidia.js

```diff
@@ -0,0 +1,42 @@
+const { Client, GatewayIntentBits } = require('discord.js');
+const { commands, commandPrefix } = require('../configs');
+require('dotenv').config();
+
+class Tadidia {
+    constructor() {
+        this.client = new Client({ intents: [GatewayIntentBits.MessageContent, GatewayIntentBits.Guilds, GatewayIntentBits.GuildMessages] });
+        this.init();
+    }
+
+    login = () => {
+        this.client.login(process.env.LOGIN_TOKEN);
+    }
+
+    getReady = () => {
+        this.client.on('ready', () => {
+            console.log(`Logged in as ${this.client.user.tag}`);
+        });
+    }
+
+    findCommand = (message) => {
+        return commands.find(command =>
+            message.content.startsWith(commandPrefix + command.text)
+        );
+    }
+
+    messageListener = () => {
```

Comment:
this method could be an `addListener`, receiving a string `event` and function `processor` and dynamically add listeners. it can also be used for other methods in this class